File: frontend/entities/ParticipantEvent.ts
import MinutesAdded from "./MinutesAdded";
import RaffleItem from "./RaffleItem";
import ActivityEvent from "./ActivityEvent";


export default class ParticipantEvent {
    id: string;
    anonymous: boolean;
    minutesAddeds: MinutesAdded[];
    raffleItems: RaffleItem[];
    activityEvent: ActivityEvent | null;
    activityEventId: string

    constructor(
        id: string,
        anonymous: boolean,
        minutesAddeds: MinutesAdded[],
        raffleItems: RaffleItem[],
        activityEvent: ActivityEvent | null,
        activityEventId: string,
    ) {
        this.id = id;
        this.anonymous = anonymous;
        this.minutesAddeds = minutesAddeds;
        this.raffleItems = raffleItems;
        this.activityEvent = activityEvent;
        this.activityEventId = activityEventId
    }
}

File: backend/App.Domain/ParticipantEvent.cs
using App.Domain.Identity;
using Base.Domain;

namespace App.Domain;

public class ParticipantEvent : BaseEntityId
{
    public Guid AppUserId { get; set; }
    public AppUser? AppUser { get; set; }
    public bool Anonymous { get; set; }
    
    public ICollection<MinutesAdded>? MinutesAddeds { get; set; }
    public ICollection<RaffleItem>? RaffleItems { get; set; }
    
    public ActivityEvent? ActivityEvent { get; set; }
    public Guid ActivityEventId { get; set; }
}